esp-wifi/src/compat/timer_compat.rs

```diff
@@ -33,17 +38,102 @@ pub(crate) struct Timer {
     pub active: bool,
     pub periodic: bool,
     pub callback: TimerCallback,
+
+    next: *mut Timer,
 }
 
 impl Timer {
     pub(crate) fn id(&self) -> usize {
         self.ets_timer as usize
     }
 }
-#[cfg(coex)]
-pub(crate) static mut TIMERS: heapless::Vec<Timer, 34> = heapless::Vec::new();
-#[cfg(not(coex))]
-pub(crate) static mut TIMERS: heapless::Vec<Timer, 20> = heapless::Vec::new();
+
+pub(crate) struct TimerQueue {
+    head: *mut Timer,
```

Comment:
Can this be `Option<Timer>`? (No boxing, first timer can be inline IMO)

Reply:
mhhh - would add even more "specials" for the head of the list - I think the code is easier with `Option<Box<Timer>>`

Reply:
I'm not sure I want to accept this reasoning without seeing how it would look like, but the current state looks good enough as is.